src/main/kotlin/org/neo4j/graphql/Predicates.kt

```diff
@@ -81,23 +98,48 @@ data class IsNullPredicate(val fieldName: String, val op: RelationOperator, val
     }
 }
 
-data class ExpressionPredicate(val name: String, val op: FieldOperator, val value: Any?, val fieldDefinition: GraphQLFieldDefinition) : Predicate {
+data class IsNullFieldPredicate(val fieldName: String, val op: FieldOperator, val type: GraphQLFieldsContainer) : Predicate {
+    override fun toExpression(variable: String): Cypher {
+        val check = if (op == FieldOperator.IS_NOT_NULL) "IS NOT NULL" else "IS NULL"
+        return Cypher("$variable.$fieldName $check")
+    }
+}
+
+data class ExpressionPredicate(
+        val name: String,
+        val nestedField: String? = null,
+        val op: FieldOperator,
+        val value: Any?,
+        val fieldDefinition: GraphQLFieldDefinition
+) : Predicate {
     val not = if (op.not) "NOT " else ""
     override fun toExpression(variable: String): Cypher {
-        val paramName: String = ProjectionBase.FILTER + paramName(variable, name, value).capitalize() + "_" + op.name
+        var paramName: String = ProjectionBase.FILTER + paramName(variable, name, value).capitalize() + "_" + op.name
         val query = if (fieldDefinition.isNativeId()) {
             if (op.list) {
                 "${not}ID($variable) ${op.op} [id IN \$$paramName | toInteger(id)]"
             } else {
                 "${not}ID($variable) ${op.op} toInteger(\$$paramName)"
             }
         } else {
-            "$not$variable.${name.quote()} ${op.op} \$$paramName"
+            val appendField = if (nestedField != null) {
+                paramName += "_$nestedField"
+                ".$nestedField"
```

Comment:
paramNames can't contain dots ?
is there a better way than modfiying the paramName?

Reply:
they can, but it breaks the "style" so I've chosen to use underscore

Reply:
they can't, since neo4j will try to resolve everything behind the `.` as property of a scalar type